textattack/constraints/semantics/thought_vector.py

```diff
@@ -0,0 +1,76 @@
+import functools
+import torch
+
+from textattack.shared import utils
+from textattack.constraints import Constraint
+from textattack.shared import WordEmbedding
+from textattack.shared import TokenizedText
+
+class ThoughtVector(Constraint):
```

Comment:
why can't this extend SentenceEncoder?

Reply:
It's kind of a dumb way of encoding a sentence but it is one right?

Reply:
good idea